server/controllers/bookController.js

```diff
@@ -90,6 +90,50 @@ class BookController {
       Utils.errorStatus(res, 500, error.message);
     }
   }
+
+  static async lendBook(req, res) {
+    const { isbn, title, patronId, cost} = req.body;
+    
+    try {
+      const booksNotReturned = await models.Borrowed_books.findAndCountAll({ where: { patronId, returned : false } });
+
+      if (booksNotReturned.count >= 3) {
+        return Utils.errorStatus(res, 405, 'You cannot have more than 3 books in your possession');        
+      } 
+
+      const today = new Date();
+      const overdueBook = booksNotReturned.rows.find((book) => book.dueDate < today);
+
+      if(overdueBook) {
+        const dueDate = overdueBook.dueDate.toString().substring(0, 15);
+        const responseObj = {
+          message: `The book titled ${overdueBook.title.toUpperCase()} was due for return on ${dueDate}. Kindly return before we can proceed with a new request`
+        }
+        return Utils.errorStatus(res, 405, responseObj); 
+      }
+
+      const dateBorrowed = new Date();
+      const dueDate = new Date();
+      // Books are due for return after 3 days
+      dueDate.setDate(dueDate.getDate() + 3); 
+
+      const borrowedBook = await models.Borrowed_books.create({
+        isbn,
+        title,
+        dateBorrowed,
+        dueDate,
+        patronId,
+        cost
```

Comment:
I think you should also remove this cost because it might conflict with the real amount on the book's table.

P.s: Abraham said we should make the platform a monthly payment, Books would no longer be paid for again. All the patron pays for is their monthly bill, damages to book, and overdue books.

Reply:
Noted bro, will work on it soon. Thanks